test/test_jit_expression.py

```diff
@@ -141,3 +142,162 @@ def test_rank1(compile_args):
     u_correct = np.array([f[1], f[0]]) + gradf0
 
     assert np.allclose(u_ffcx, u_correct.T)
+
+
+def test_elimiate_zero_tables_scalar(compile_args):
```

Comment:
As all will fo through same codepath, I think it suffices to have the tensor test.